internal/http/client.go

```diff
@@ -122,12 +122,18 @@ func (c *NewRelicClient) Put(
 // The respBody argument will be unmarshaled from JSON in the response body to the type provided.
 // If respBody is not nil and the response body cannot be unmarshaled to the type provided, an error will be returned.
 func (c *NewRelicClient) Delete(url string,
-	queryParams *map[string]string,
+	queryParams *[]QueryParam,
 	respBody interface{},
 ) (*http.Response, error) {
 	return c.do(http.MethodDelete, url, queryParams, nil, respBody)
 }
 
+// QueryParam is used to pass query params to one of the client's request methods.
+type QueryParam struct {
+	Name  string
```

Comment:
Thoughts on changing this field to `Key`? The main reason I ask is because Go docs uses "key" and "value" terminology in the [corresponding documentation](https://golang.org/src/net/url/url.go?s=23981:24012#L816). 

The word `Key` also seems a little more clear to me vs `Name` which kind of feels more "value-like" if that makes any sense.